conan/tools/cmake/cmakedeps2/target_configuration.py

```diff
@@ -81,13 +81,16 @@ def _requires(self, info, components):
                         assert required_pkg == required_comp
                         comp = None
                         default_target = f"{dep.ref.name}::{dep.ref.name}"  # replace_requires
+                        link = pkg_type is not PackageType.SHARED 
```

Comment:
In the test that I added, this movement is not necessary.
If this is necessary for a different use case, it would be great to have yet another test that covers this scenario.

Reply:
You're right this is triggered by another use case, see new test added. That triggers the first condition dep_comp == None, and then crash when evaluating the link. 

The fix is my own looked a simple way to evaluate link in that case from my understanding of the feature feel free to provide another fix.

Reply:
Ok, now with the test and the explanation is more clear, many thanks!
I have just fixed an indent error in the test, and CI is running.